zap/src/main/java/org/parosproxy/paros/network/HttpSender.java

```diff
@@ -615,12 +655,17 @@ private HttpMethod runMethod(HttpMessage msg, boolean isFollowRedirect, HttpMeth
             // cant do this for EntityEnclosingMethod methods - it will fail
             method.setFollowRedirects(isFollowRedirect);
         }
+
+        // needs to verify if the state
+        // should store cookies
+        updateCookieState();
```

Comment:
Is this really needed? The state/settings should be properly set always?

Reply:
I think it isn't needed, as per your comment on not using `client.getState()` it was probably only necessary because the cookies were set for the request and needed to be deleted by resetting the state.